The "Quarkus development mode" instructions in quarkus/README.md throw a ForkJoinPool error

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

The development mode instructions in quarkus/README.md throw this exception:

```
2024-06-08 15:49:56,942 ERROR [org.keycloak.quarkus.runtime.KeycloakMain] (Quarkus Main Thread) The ForkJoinPool has been initialized with the wrong thread factory. The property 'java.util.concurrent.ForkJoinPool.common.threadFactory' should be set on the Java CLI to ensure Java's ForkJoinPool will always be initialized with 'null' even if there are Java agents which might initialize logging or other capabilities earlier than the main method.
2024-06-08 15:49:56,942 ERROR [io.quarkus.runner.bootstrap.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:113)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: The ForkJoinPool has been initialized with the wrong thread factory
        at org.keycloak.quarkus.runtime.KeycloakMain.ensureForkJoinPoolThreadFactoryHasBeenSetToQuarkus(KeycloakMain.java:117)
        at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:60)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 3 more
```

### Version

86101b3d1d8af4080a1e07b91cb64425b1e607cd

### Regression

- [ ] The issue is a regression

### Expected behavior

I expect that the dev server command in the documentation should work.

### Actual behavior

```
[INFO] --- compiler:3.8.1:compile (default-compile) @ keycloak-quarkus-server-app ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- quarkus:3.8.5:dev (default-cli) @ keycloak-quarkus-server-app ---
[INFO] Invoking resources:3.2.0:testResources (default-testResources) @ keycloak-quarkus-server-app
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory /home/jade/src/co/keycloak/quarkus/server/src/test/resources
[INFO] Invoking compiler:3.8.1:testCompile (default-testCompile) @ keycloak-quarkus-server-app
[INFO] No sources to compile
[WARNING] [io.quarkus.bootstrap.resolver.maven.workspace.WorkspaceLoader] Module(s) under /home/jade/src/co/keycloak/testsuite/integration-arquillian/tests/other/adapters/jboss/relative will be handled as thirdparty dependencies because /home/jade/src/co/keycloak/testsuite/integration-arquillian/tests/other/adapters/jboss/relative/pom.xml does not exist
[WARNING] [io.quarkus.bootstrap.devmode.DependenciesFilter] Live reload was disabled for the following project artifacts:
- org.keycloak:keycloak-quarkus-server:999.0.0-SNAPSHOT
- org.keycloak:keycloak-services:999.0.0-SNAPSHOT
- org.keycloak:keycloak-js-adapter-jar:999.0.0-SNAPSHOT
- org.keycloak:keycloak-common:999.0.0-SNAPSHOT
- org.keycloak:keycloak-core:999.0.0-SNAPSHOT
- org.keycloak:keycloak-crypto-default:999.0.0-SNAPSHOT
- org.keycloak:keycloak-crypto-fips1402:999.0.0-SNAPSHOT
- org.keycloak:keycloak-server-spi:999.0.0-SNAPSHOT
- org.keycloak:keycloak-server-spi-private:999.0.0-SNAPSHOT
- org.keycloak:keycloak-themes:999.0.0-SNAPSHOT
- org.keycloak:keycloak-saml-core-public:999.0.0-SNAPSHOT
- org.keycloak:keycloak-saml-core:999.0.0-SNAPSHOT
- org.keycloak:keycloak-model-jpa:999.0.0-SNAPSHOT
- org.keycloak:keycloak-model-infinispan:999.0.0-SNAPSHOT
- org.keycloak:keycloak-model-storage:999.0.0-SNAPSHOT
- org.keycloak:keycloak-model-storage-private:999.0.0-SNAPSHOT
- org.keycloak:keycloak-model-storage-services:999.0.0-SNAPSHOT
- org.keycloak:keycloak-authz-policy-common:999.0.0-SNAPSHOT
- org.keycloak:keycloak-kerberos-federation:999.0.0-SNAPSHOT
- org.keycloak:keycloak-sssd-federation:999.0.0-SNAPSHOT
- org.keycloak:keycloak-ldap-federation:999.0.0-SNAPSHOT
- org.keycloak:keycloak-config-api:999.0.0-SNAPSHOT
- org.keycloak:keycloak-account-ui:999.0.0-SNAPSHOT
- org.keycloak:keycloak-admin-ui:999.0.0-SNAPSHOT
- org.keycloak:keycloak-rest-admin-ui-ext:999.0.0-SNAPSHOT
- org.keycloak:keycloak-quarkus-server-deployment:999.0.0-SNAPSHOT
The artifacts above appear to be either dependencies of non-reloadable application dependencies or Quarkus extensions
Listening for transport dt_socket at address: 5005
2024-06-08 15:49:56,942 ERROR [org.keycloak.quarkus.runtime.KeycloakMain] (Quarkus Main Thread) The ForkJoinPool has been initialized with the wrong thread factory. The property 'java.util.concurrent.ForkJoinPool.common.threadFactory' should be set on the Java CLI to ensure Java's ForkJoinPool will always be initialized with 'null' even if there are Java agents which might initialize logging or other capabilities earlier than the main method.
2024-06-08 15:49:56,942 ERROR [io.quarkus.runner.bootstrap.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:113)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: The ForkJoinPool has been initialized with the wrong thread factory
        at org.keycloak.quarkus.runtime.KeycloakMain.ensureForkJoinPoolThreadFactoryHasBeenSetToQuarkus(KeycloakMain.java:117)
        at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:60)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 3 more
```

### How to Reproduce?

1. Get clean checkout of keycloak; run `git clean -fxd`, or so.
2. Follow "building the project the first time" section in quarkus/README.md:
```
    ../mvnw -f ../pom.xml clean install -DskipTestsuite -DskipExamples -DskipTests
```
3. Run the command from the "Contributing" section of the same file:
```
    ../mvnw -f server/pom.xml compile quarkus:dev -Dquarkus.args="start-dev"
```

Get the exception shown above, and then:

```
2024-06-08 15:49:57,119 INFO  [io.quarkus.deployment.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2024-06-08 15:51:10,616 INFO  [io.quarkus.deployment.dev.RuntimeUpdatesProcessor] (Aesh InputStream Reader) Instrumentation based restart enabled

--
Tests paused
Press [e] to edit command line args (currently 'start-dev'), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>
```

### Anything else?

This is on a:

openjdk 21.0.3 2024-04-16
OpenJDK Runtime Environment (build 21.0.3+9)
OpenJDK 64-Bit Server VM (build 21.0.3+9, mixed mode, sharing)

On Arch Linux.

--- a/quarkus/server/pom.xml
+++ b/quarkus/server/pom.xml
@@ -62,6 +62,7 @@
                     <finalName>keycloak</finalName>
                     <systemProperties>
                         <kc.home.dir>${kc.home.dir}</kc.home.dir>
+                        <java.util.concurrent.ForkJoinPool.common.threadFactory>io.quarkus.bootstrap.forkjoin.QuarkusForkJoinWorkerThreadFactory</java.util.concurrent.ForkJoinPool.common.threadFactory>
                     </systemProperties>
                 </configuration>
                 <executions>

